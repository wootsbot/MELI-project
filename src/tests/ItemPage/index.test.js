import { render, screen, act } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import ItemsPage from '@/pages/[items]/[itemId]';

const renderComponent = (props = {}) => render(<ItemsPage {...props} />);

const useRouter = jest.spyOn(require('next/router'), 'useRouter');

const itemMock = {
  id: 'MLA903951222',
  site_id: 'MLA',
  title: 'Cable Cargador Usb Original Lightning iPhone 6 7 8 X Xs Xr',
  price: 1899.99,
  condition: 'new',
  thumbnail: '/D_659328-MLA47006823834_082021-O.jpg',
  sold_quantity: 3,
  condition: 'new',
  address: {
    state_id: 'AR-C',
    state_name: 'Capital Federal',
    city_id: null,
    city_name: 'Belgrano',
  },
  shipping: {
    free_shipping: true,
    mode: 'me2',
    tags: ['self_service_in'],
    logistic_type: 'cross_docking',
    store_pick_up: false,
  },
  attributes: [
    { id: 'BRAND', value_name: 'name' },
    { id: 'MODEL', value_name: 'name' },
  ],
};

jest.mock('../../hooks/useItem', () => {
  return jest.fn(() => ({
    item: itemMock,
    itemDescription: { plain_text: 'description' },
  }));
});

jest.mock('../../hooks/useDomainDiscovery', () => {
  return jest.fn(() => ({
    discovery: { category_name: 'category name', attributes: [{ id: '123', value_name: 'name' }] },
  }));
});

describe('<ItemPage />', () => {
  beforeEach(() => {
    jest.unmock('../../hooks/useItems');
    jest.unmock('../../hooks/useDomainDiscovery');
    jest.resetModules();
  });

  it('Should render correctly', () => {
    const router = { push: jest.fn(), query: { search: null } };
    useRouter.mockReturnValue(router);

    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });

  it('should render properties of mock item when render page', () => {
    const router = { push: jest.fn(), query: { search: 'iphone' } };

    useRouter.mockReturnValue(router);

    renderComponent();

    const conditionElement = screen.getByText(/nuevo \- 3 vendidos/i);
    const headingH1 = screen.getByRole('heading', {
      name: /cable cargador usb original lightning iphone 6 7 8 x xs xr/i,
    });
    const priceElement = screen.getByText(/\$ 1,899\.99/i);
    const buttonPay = screen.getByRole('button', {
      name: /comprar/i,
    });
    const descriptionElement = screen.getByRole('heading', {
      name: /descripci√≥n del producto/i,
    });

    expect(conditionElement).toBeInTheDocument();
    expect(headingH1).toBeInTheDocument();
    expect(priceElement).toBeInTheDocument();
    expect(buttonPay).toBeInTheDocument();
  });
});
