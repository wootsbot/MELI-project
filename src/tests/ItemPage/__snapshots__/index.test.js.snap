// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemPage /> Should render correctly 1`] = `
<div>
  <div
    class="root"
  >
    <div
      class="margin-xs-16"
    >
      <nav
        aria-label="breadcrumb"
      >
        <ol
          class="ol"
        >
          <li
            class="li"
          >
            <span>
              category name
            </span>
          </li>
          <li
            aria-hidden="true"
            class="separator"
          >
            &gt;
          </li>
          <li
            class="li"
          >
            <span>
              name
            </span>
          </li>
          <li
            aria-hidden="true"
            class="separator"
          >
            &gt;
          </li>
          <li
            class="li"
          >
            <span>
              name
            </span>
          </li>
        </ol>
      </nav>
    </div>
    <div
      class="margin-xs-16--top"
    >
      <div
        class="margin-xs-32--bottom"
      >
        <div
          class="root"
        >
          <div
            class="details"
          >
            <div
              class="thumbnail-wrapper--lg"
            >
              <div
                style="overflow: hidden; box-sizing: border-box; display: inline-block; position: relative; width: 680px; height: 680px;"
              >
                <noscript />
                <img
                  alt="Cable Cargador Usb Original Lightning iPhone 6 7 8 X Xs Xr"
                  class="thumbnail"
                  data-nimg="true"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
              </div>
            </div>
            <div
              class="thumbnail-wrapper--sm"
            >
              <div
                style="overflow: hidden; box-sizing: border-box; display: inline-block; position: relative; width: 400px; height: 400px;"
              >
                <noscript />
                <img
                  alt="Cable Cargador Usb Original Lightning iPhone 6 7 8 X Xs Xr"
                  class="thumbnail"
                  data-nimg="true"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
              </div>
            </div>
            <div>
              <div
                class="margin-xs-16--bottom header"
              >
                <span>
                  Nuevo - 3 vendidos
                </span>
              </div>
              <h1
                class="title"
              >
                Cable Cargador Usb Original Lightning iPhone 6 7 8 X Xs Xr
              </h1>
              <div
                class="margin-xs-32"
              >
                <span
                  class="price"
                >
                  $ 1,899.99
                </span>
              </div>
              <button
                class="button button--primary"
              >
                <span
                  class="button--label"
                >
                  Comprar
                </span>
              </button>
            </div>
          </div>
          <div>
            <div
              class="margin-xs-32--bottom"
            >
              <h2
                class="description-title"
              >
                Descripci√≥n del producto
              </h2>
            </div>
            <p
              class="description"
            >
              description
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
